Force binary mode for open() on WIN32

kinichiro opened the pull request:
This PR is for #542 and just adds O_BINARY for open() on WIN32 platform.
Should we solve this by other way rather than by patch as this PR ?
Thoughts ?

busterb commented:
Is there any reason not to just convert all `open `calls via the posix intercept layer we have to be unconditionally binary-only for consistency? I think we already do this with our `posix_open` intercept in posix_win.c in libcrypto.

kinichiro commented:
Yes, we can use posix_open().
I thought posix_win.c should adapt only for winsock operations, but it was my misunderstood.
I added another commit that removes 3 patches that kept from using posix_*() functions.
I also modified posix_win.c to handle WSANOTINITIALISED that happens when WSAStartup() is not called.
I noticed that keypairtest fails since it got WSANOTINITIALISED in posix_read().

ok?
